src/client/app/effects/snippet.ts

```diff
@@ -372,14 +360,41 @@ export class SnippetEffects {
         properties['mode'] = isViewMode ? 'view' : 'editor';
         AI.trackEvent(mode, properties);
 
-        /*
-         * If a imported snippet is a SAMPLE or the app is in view mode, then skip the save (simply to avoid clutter).
-         * The snippet will get saved as soon as the user makes any changes (if in editor mode).
+        /**
+         * If the action here involves true importing rather than re-opening,
+         * and if the name is already taken by a local snippet, ask the user if they want to:
+         * import the snippet with a new name, go to the already existing snippet, or cancel the import
          */
-        if (mode !== Snippet.ImportType.SAMPLE && !isViewMode) {
-            actions.push(new Snippet.SaveAction(snippet));
+        let importResult = null;
+        if (mode !== Snippet.ImportType.OPEN && this._exists(snippet.name)) {
```

Comment:
This has been the existing logic since Script Lab's initial deployment; when we imported a snippet, we checked it against your existing snippets' names and then added the suffix if there was a name conflict. It made sense to me since otherwise your snippet list would grow confusing. We now want to change this logic entirely?